From ed24a12bd7d258b62e8924540233660eba618c0b Mon Sep 17 00:00:00 2001
From: jayanth1 <shankar.srinivas.jayanthi@intel.com>
Date: Wed, 14 Oct 2020 21:08:07 +0530
Subject: [PATCH] 
 apl_nuc-Fix-for-Pairing-re-initiated-after-Bluetooth-Reset.patch for r41

Tracked-On: OAM-94320
Signed-off-by: jayanth1 <shankar.srinivas.jayanthi@intel.com>

diff --git a/src/com/android/bluetooth/btservice/AdapterService.java b/src/com/android/bluetooth/btservice/AdapterService.java
index a65bcfdf9..f85e828a2 100644
--- a/src/com/android/bluetooth/btservice/AdapterService.java
+++ b/src/com/android/bluetooth/btservice/AdapterService.java
@@ -1559,6 +1559,12 @@ public class AdapterService extends Service {
             if (service == null) {
                 return false;
             }
+            BluetoothDevice[] bondedDeviceList = getBondedDevices();
+            if (bondedDeviceList != null) {
+                for (BluetoothDevice device : bondedDeviceList) {
+                    service.removeBond(device);
+                }
+            }
             return service.factoryReset();
 
         }
-- 
2.28.0

